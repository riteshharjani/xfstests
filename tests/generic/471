#! /bin/bash
# SPDX-License-Identifier: GPL-2.0
# Copyright (c) 2023 IBM Corporation.  All Rights Reserved.
#
# FS QA Test 471
#
# Integrity test for O_SYNC with buff-io, dio, aio-dio with sudden shutdown
#
. ./common/preamble
_begin_fstest auto quick shutdown aio

# real QA test starts here
_supported_fs generic
_require_scratch
_require_scratch_shutdown
_require_odirect
_require_aiodio aio-dio-write-verify

_scratch_mkfs > $seqres.full 2>&1
_scratch_mount

echo "T-1: Create a 1M file using buff-io & O_SYNC"
$XFS_IO_PROG -fs -c "pwrite -S 0x5a 0 1M" $SCRATCH_MNT/testfile.t1 > /dev/null 2>&1
echo "T-1: Shutdown the fs suddenly"
_scratch_shutdown
echo "T-1: Cycle mount"
_scratch_cycle_mount
echo "T-1: File contents after cycle mount"
_hexdump $SCRATCH_MNT/testfile.t1

echo "T-2: Create a 1M file using O_DIRECT & O_SYNC"
$XFS_IO_PROG -fsd -c "pwrite -S 0x5a 0 1M" $SCRATCH_MNT/testfile.t2 > /dev/null 2>&1
echo "T-2: Shutdown the fs suddenly"
_scratch_shutdown
echo "T-2: Cycle mount"
_scratch_cycle_mount
echo "T-2: File contents after cycle mount"
_hexdump $SCRATCH_MNT/testfile.t2

echo "T-3: Create a 1M file using AIO-DIO & O_SYNC"
$AIO_TEST -a size=1048576 -S -N $SCRATCH_MNT/testfile.t3 > /dev/null 2>&1
echo "T-3: Shutdown the fs suddenly"
_scratch_shutdown
echo "T-3: Cycle mount"
_scratch_cycle_mount
echo "T-3: File contents after cycle mount"
_hexdump $SCRATCH_MNT/testfile.t3

status=0
exit
