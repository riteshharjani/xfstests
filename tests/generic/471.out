QA output created by 471
T-1: Create a 1M file using buff-io & O_SYNC
T-1: Shutdown the fs suddenly
T-1: Cycle mount
T-1: File contents after cycle mount
000000 5a 5a 5a 5a 5a 5a 5a 5a 5a 5a 5a 5a 5a 5a 5a 5a  >ZZZZZZZZZZZZZZZZ<
*
100000
T-2: Create a 1M file using O_DIRECT & O_SYNC
T-2: Shutdown the fs suddenly
T-2: Cycle mount
T-2: File contents after cycle mount
000000 5a 5a 5a 5a 5a 5a 5a 5a 5a 5a 5a 5a 5a 5a 5a 5a  >ZZZZZZZZZZZZZZZZ<
*
100000
T-3: Create a 1M file using AIO-DIO & O_SYNC
T-3: Shutdown the fs suddenly
T-3: Cycle mount
T-3: File contents after cycle mount
000000 5a 5a 5a 5a 5a 5a 5a 5a 5a 5a 5a 5a 5a 5a 5a 5a  >ZZZZZZZZZZZZZZZZ<
*
100000
